tmt/steps/prepare/distgit.py

```diff
@@ -73,7 +73,6 @@ def insert_to_prepare_step(
     future_requires: PreparePlugin[Any] = cast(
         PreparePlugin[Any], PreparePlugin.delegate(prepare_step, data=data_require)
     )
```

Comment:
I am thinking we could move the `StepPlugin.delegate` inside the add_phase as a
```python
@overload
def add_phase(self, data: StepData) -> None:
@overload
def add_phase(self, phase: Phase) -> None:

def add_phase(self, step_or_data) -> None:
    if isinstance(step_or_data, StepData):
        phase = self._plugin_base_class.delegate(self, data=setp_or_data)
```

Reasoning for this is that on the caller side, it is more natural (and sufficient) to interact with the `StepData` as surrogates for the phases, i.e. it is closer to them writing the yaml file and sticking it into the plan.

Reply:
Hmmmm, probably later? Because I would really, really like to add some type checking here. I can see this allowing `plan.prepare.add_phase(ReportPlugin())`, without step turning `Generic` & parameterize types of its step data and its phases; the more work we move into helpers like `add_phase()`, the less comfortable I would feel about relaxing annotations we have in place. Since you started poking this area, I guess it's time to refresh my git stash and get it done to make checking stronger.